src/core/treeNodeValue.ts

```diff
@@ -224,12 +263,48 @@ export interface TreeNodeValueOptions extends ParseSegmentOptions {
 export function createTreeNodeValue(
   segment: string,
   parent?: TreeNodeValue,
-  options: TreeNodeValueOptions = {}
+  opts: TreeNodeValueOptions = {}
 ): TreeNodeValue {
   if (!segment || segment === 'index') {
     return new TreeNodeValueStatic(segment, parent, '')
   }
 
+  // ensure default options
+  const options = resolveTreeNodeValueOptions(opts)
+
+  // extract the group between parentheses
+  const openingPar = segment.indexOf('(')
+
+  if (options.format === 'file' && openingPar >= 0) {
+    let groupName: string
+    let groupPathSegment: string
+
+    const closingPar = segment.lastIndexOf(')')
+    if (closingPar < 0 || closingPar < openingPar) {
+      throw new Error(`Invalid segment: "${segment}"`)
+    }
+
+    groupName = segment.slice(openingPar + 1, closingPar)
+    const before = segment.slice(0, openingPar)
+    const after = segment.slice(closingPar + 1)
+
+    if (!before && !after) {
+      // pure group: no contribution to the path
+      groupPathSegment = ''
+    } else if (!before) {
+      // no leading static part, so just use what's after
+      groupPathSegment = after
+    } else if (!after) {
+      // no trailing static part, so just use what's before
+      groupPathSegment = before
+    } else {
+      // both before and after exist, separate them with a dash
```

Comment:
@posva But if he doesn't add a dash, he will end up with `userssearch.vue`. Is that okay? That was my main reasoning behind this (also, it's ugly :grin:)
On the other hand, it can be convenient not to enforce dash for a case where user wants to use camelCase for  naming:
example: users-(details)Search => usersSearch. Adding dashes would prevent them to have camelCase in this case

Reply:
Yes, that's okay. They have to put the dash, it's more flexible this way. If they want to rename paths, they can use the `extendRoute` option

Reply:
@posva done [commit](https://github.com/posva/unplugin-vue-router/pull/549/commits/f372943ce50ccb3fa062c5bc8502381c9236f9f3)